app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt

```diff
@@ -64,12 +73,68 @@ class EditPatientScreen(context: Context, attributeSet: AttributeSet) : Relative
   @field:[Inject Named("date_for_user_input")]
   lateinit var dateOfBirthFormat: DateTimeFormatter
 
+  @Inject
+  lateinit var numberValidator: PhoneNumberValidator
+
+  @Inject
+  lateinit var dateOfBirthValidator: UserInputDateValidator
+
+  @Inject
+  lateinit var userClock: UserClock
+
+  @Inject
+  lateinit var patientRepository: PatientRepository
+
+  @Inject
+  lateinit var utcClock: UtcClock
+
+  @Inject
+  lateinit var schedulersProvider: SchedulersProvider
+
   @Inject
   lateinit var activity: TheActivity
 
   @Inject
   lateinit var crashReporter: CrashReporter
 
+  private val screenKey by unsafeLazy {
+    screenRouter.key<EditPatientScreenKey>(this)
+  }
+
+  private val viewRenderer by unsafeLazy {
+    EditPatientViewRenderer(this)
+  }
+
+  private val events: Observable<EditPatientEvent>
+    get() = Observable.mergeArray(
+        saveClicks(),
+        nameTextChanges(),
+        phoneNumberTextChanges(),
+        districtTextChanges(),
+        stateTextChanges(),
+        colonyTextChanges(),
+        genderChanges(),
+        dateOfBirthTextChanges(),
+        dateOfBirthFocusChanges(),
+        ageTextChanges(),
+        backClicks()
+    ).compose(ReportAnalyticsEvents())
+        .cast()
+
+  private lateinit var eventsDisposable: Disposable
+
+  private val delegate by unsafeLazy {
+    val (patient, address, phoneNumber) = screenKey
+    MobiusDelegate(
```

Comment:
We should look at directly injecting the delegate using `@AssistedInject` so that the key can be provided at some point. What do you think?

Reply:
I don't know what `@AssistedInject` is, can we discuss this in person?

Reply:
Sure. But the tl;dr is that [Assisted Injection](https://github.com/square/AssistedInject) will allow components like the screens to contribute dependencies to the dagger graph (like the screen key in this case) when injecting it.

Anyway, it's not important to do in this PR. Just adding that it might be something that will allow us to inject mobius delegates into screens easily.